### PR TITLE
Conseil 230 v2 query bug fixes

### PR DESCRIPTION
**Pull request by piotrkosecki**
Things I've fixed/added:
- api key verification
- platform verification
- added docs with each operation example
- query and predicate fields validation
- extended values left for sql so value could represent timestamp in ISO8601 format

### CODE REVIEW

File: src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala [hunk 1]
```diff
@@ -40,13 +40,14 @@ trait RouteHandling {
...
     * @return Either validated query or bad request if validation failed
     */
-  protected def validateQueryOrBadRequest(fieldQuery: Query): Directive1[Query] = Directive { inner =>
-    fieldQuery.validate match {
-      case Some(value) => inner(Tuple1(value))
-      case None => complete(StatusCodes.BadRequest)
+  protected def validateQueryOrBadRequest(entity: String, fieldQuery: ApiQuery): Directive1[Query] = Directive { inner =>
+    fieldQuery.validate(entity) match {
+      case Right(value) => inner(Tuple1(value))
+      case Left(errors) => complete(StatusCodes.BadRequest -> s"Errors: ${errors.mkString(",")}")
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala]: Did we consider returning a json content to parse even when there are errors?

Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala]: Possibly using `mkString` with a `\n` separator might be more readable as text

File: src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala [hunk 2]
```diff
@@ -62,6 +63,15 @@ trait RouteHandling {
     }
     complete(response)
   }
+
+  /** validates platform */
+  protected def validatePlatform(config: PlatformsConfiguration, platform: String): Directive0 = {
+    getNetworks(config, platform) match {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala]: Idea: would an `if/else` looking for `isEmpty` be easier to follow?

File: src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala [hunk 3]
```diff
@@ -4,20 +4,26 @@ import com.fasterxml.jackson.core.`type`.TypeReference
...
+  import io.scalaland.chimney.dsl._
 
   /** Default value of limit parameter */
   val defaultLimitValue: Int = 10000
 
   /** Max value of limit parameter */
   val maxLimitValue: Int = 100000
 
+  /** Trait representing query validation errors */
+  sealed trait QueryValidationError {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: extending Product with Serializable might be useful later on, for type inference

File: src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala (hunk 1 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala]: ok, I've added `\n`

File: src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala (hunk 2 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/util/RouteHandling.scala]: I've changed to `if/else`

File: src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala (hunk 3 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: sure, added

File: src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala [hunk 4]
```diff
@@ -1,5 +1,7 @@
 package tech.cryptonomic.conseil.generic.chain
 
+import java.io.Serializable
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: no need, there's an alias in `_root_.scala` package

File: src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala [hunk 5]
```diff
@@ -47,8 +49,7 @@ object DataTypes {
   /** Class representing invalid predicate field */
   case class InvalidPredicateField(message: String) extends QueryValidationError
 
-  /** Class representing unexpected error */
-  case class UnexpectedError(message: String) extends QueryValidationError
+  case class ErrorFields()
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: 🤔 ?

Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: removed this ;)

File: src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala (hunk 4 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala]: I don't know why intellij did not see `Serializable` at that moment but you are right, there it is :)

**Pull request merged**